src/impl/collection_notifier.hpp

```diff
@@ -211,7 +213,7 @@ class CollectionNotifier {
 
     // Iteration variable for looping over callbacks
     // remove_callback() updates this when needed
-    size_t m_callback_index = npos;
+    size_t m_callback_index = -1;
```

Comment:
realm::npos is defined in `array.hpp` which isn't actually supposed to be a public header, and the code using it relies on npos being -1.